students/EricAdams/session07/html_render.py

```diff
@@ -15,18 +17,21 @@ def append(self, content):
         self.content.append(content)
 
     def render(self, file_obj, indent=''):
-        # all_content = ('<' + self.tag + '>')
-        html_tag = ('<' + self.tag + '>')
+        # what recursion level are we going in
+        Element.depth += 1
```

Comment:
problem here -- every single Element will now have a larger recursion depth!

If this is working, then it's because you are changing it constantly to keep in sync.

But better to simply pass it through when you render.

That's what the indent parameter was for -- to pas in the current level of indentation -- you could pass in a "depth" instead. but better to pass it through that store it in the Element class.